### PR TITLE
 google and mail auth bloc

### PR DESCRIPTION
**Pull request by vaishnavi-2307**

### CODE REVIEW

File: lib/src/app/feature/authentication/presentation/login_screen.dart [hunk 1]
```diff
@@ -156,116 +160,184 @@ class _LoginScreenState extends State<LoginScreen> {
...
+                              child: Row(
+                                mainAxisAlignment: MainAxisAlignment.center,
+                                crossAxisAlignment: CrossAxisAlignment.center,
+                                children: [
+                                  Flexible(
+                                    flex: 2,
+                                    child: FittedBox(
+                                      child: Material(
+                                        child: InkWell(
+                                          onTap: () async {
```
Comment (japatel225) [inline lib/src/app/feature/authentication/presentation/login_screen.dart:228]: I see `async` is not added in this PR and is not related to the task of this PR, but can we remove it if not required? Or should we do such unrelated "clean-up" changes (this one and others if any) in separate PR ideally? Let me know your thoughts.

File: lib/src/app/feature/authentication/presentation/login_screen.dart [hunk 2]
```diff
@@ -156,116 +160,184 @@ class _LoginScreenState extends State<LoginScreen> {
...
-            ],
-          ),
-        ),
-      ),
+              );
+            } else if (state is AuthLoading) {
+              return const Center(child: CircularProgressIndicator());
+            }
+            return const Center(
+                child:SizedBox.shrink());
```
Comment (japatel225) [inline lib/src/app/feature/authentication/presentation/login_screen.dart:336]: same comment regarding code auto-format.

File: android/app/src/google-services.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
+{
```
Comment (japatel225) [inline android/app/src/google-services.json:1]: Can you move this file to path "android/app/google-services.json" or add it to .gitignore? So it won't be a part of the open-source repo. Users of the repo can add their own google-services.json file.

File: lib/src/app/feature/email_authentication/presentation/register.dart [hunk 4]
```diff
@@ -17,6 +17,26 @@ class _SignupState extends State<Signup> {
...
+
+    if (value == null || value.isEmpty) {
+      return 'Please Enter Password';
+    } else if (!regex.hasMatch(value)) {
+      return ("Enter Valid Password(Min. 6 Character)");
+    }
+    return null;
+  }
+
+  String? isEmailValid(String? value) {
```
Comment (japatel225) [inline lib/src/app/feature/email_authentication/presentation/register.dart:31]: Can you move this and `isPasswordValid` methods to after the build method?

**Review by japatel225 — CHANGES_REQUESTED** [review]
Please also resolve the conflicts.

**Review by Abhishek01039 — APPROVED** [review]
LGTM!

**Pull request merged**